### PR TITLE
Extend live preview to layout scripts and stylesheets

### LINKED ISSUE #1229: Extend live preview to layout scripts and stylesheets
Steps:
1. serve a site locally 
1. modify a css file inside _markbind/layouts

Problem: the changes are not reflected by the live preview

### PR DESCRIPTION
**Pull request by KendrickAng**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [ ] Documentation update
• [X] Bug fix
• [ ] New feature
• [ ] Enhancement to an existing feature
• [ ] Other, please explain:

Fixes #1229

**What is the rationale for this request?**
Layout scripts and stylesheets are currently not marked as dependencies for reload. 

**What changes did you make? (Give an overview)**
This PR adds them as dependencies, and also ensures the updated scripts/stylesheets are copied to the _site directory.

**Provide some example code that this change will affect:**

```js
// In Page/index.js
  addLayoutScriptsAndStyles() {
    this.asset.layoutScript = path.join(this.pageConfig.layoutsAssetPath, this.layout, 'scripts.js');
    this.asset.layoutStyle = path.join(this.pageConfig.layoutsAssetPath, this.layout, 'styles.css');
    this.includedFiles.add(path.join(process.cwd(), LAYOUTS_FOLDER_PATH, this.layout, 'scripts.js'));
    this.includedFiles.add(path.join(process.cwd(), LAYOUTS_FOLDER_PATH, this.layout, 'styles.css'));
  }

// In Site/index.js
this.regenerateAffectedPages(uniquePaths)
     .then(() => fs.removeAsync(this.tempPath))
     .then(() => this.copyLayouts())
     .then(resolve)
     .catch((error) => {
       // if error, remove the site and temp folders
       Site.rejectHandler(reject, error, [this.tempPath, this.outputPath]);
     });

```

**Is there anything you'd like reviewers to focus on?**

**Testing instructions:**
1. Serve markbind from the `docs` folder
2. Change scripts.js in `layouts/default` and check that the change is reflected
3. Do the same for styles.css

**Proposed commit message: (wrap lines at 72 characters)**
Extend live preview to layout scripts & stylesheets

### CODE REVIEW

File: packages/core/src/Page/index.js [hunk 1]
```diff
@@ -1165,6 +1166,8 @@ class Page {
   addLayoutScriptsAndStyles() {
     this.asset.layoutScript = path.join(this.pageConfig.layoutsAssetPath, this.layout, 'scripts.js');
     this.asset.layoutStyle = path.join(this.pageConfig.layoutsAssetPath, this.layout, 'styles.css');
+    this.includedFiles.add(path.join(process.cwd(), LAYOUTS_FOLDER_PATH, this.layout, 'scripts.js'));
```
Comment (ang-zeyu) [inline packages/core/src/Page/index.js:1169]: this is the "right" fix, but `process.cwd()` wont work when you run markbind to serve a site inside a folder
https://markbind.org/userGuide/cliCommands.html#serve-command

you want `this.rootPath`

File: packages/core/src/Site/index.js [hunk 2]
```diff
@@ -712,6 +712,7 @@ class Site {
     return new Promise((resolve, reject) => {
```
Comment (ang-zeyu) [inline packages/core/src/Site/index.js:712]: are the changes here still needed? 😮

File: packages/core/src/Page/constants.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ module.exports = {
   FOOTERS_FOLDER_PATH: '_markbind/footers',
   HEAD_FOLDER_PATH: '_markbind/head',
   HEADERS_FOLDER_PATH: '_markbind/headers',
+  LAYOUTS_FOLDER_PATH: '_markbind/layouts',
```
Comment (ang-zeyu) [inline packages/core/src/Page/constants.js:5]: there's one in `core/src/constants.js`

File: packages/core/src/Site/index.js (hunk 2 above)
Comment (KendrickAng) [reply packages/core/src/Site/index.js:712]: Sorry, these are from a debug session, I forgot to remove them. 😓

File: packages/core/src/Page/constants.js (hunk 3 above)
Comment (KendrickAng) [reply packages/core/src/Page/constants.js:5]: is there a reason why these constants are separated?

File: packages/core/src/Site/index.js (hunk 2 above)
Comment (ang-zeyu) [reply packages/core/src/Site/index.js:712]: no worries 😁

File: packages/core/src/Page/constants.js (hunk 3 above)
Comment (ang-zeyu) [reply packages/core/src/Page/constants.js:5]: Its used in `Site.js` as well

**Review by ang-zeyu — APPROVED** [review]
Looks good, thanks : )

**Pull request merged**